In [1]:
import pandas as pd
import seaborn as sns
%matplotlib inline

In [2]:
tips=pd.read_csv(r"C:\Users\KAMALI--9-9--\Downloads\Data science\datasets\tips.csv")

In [3]:
tips.shape

(244, 7)

In [4]:
tips.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [5]:
tips.columns

Index(['total_bill', 'tip', 'sex', 'smoker', 'day', 'time', 'size'], dtype='object')

In [6]:
tips.describe().transpose()

count       mean       std   min      25%     50%      75%    max
total_bill  244.0  19.785943  8.902412  3.07  13.3475  17.795  24.1275  50.81
tip         244.0   2.998279  1.383638  1.00   2.0000   2.900   3.5625  10.00
size        244.0   2.569672  0.951100  1.00   2.0000   2.000   3.0000   6.00

In [7]:
tips.isnull().any()

total_bill    False
tip           False
sex           False
smoker        False
day           False
time          False
size          False
dtype: bool

In [8]:
tips.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


# Label encoding

In [9]:
#from sklearn.preprocessing import LabelEncoder

In [10]:
#le=LabelEncoder()

In [11]:
# tips['sex']=le.fit_transform(tips['sex'])

In [12]:
# tips['smoker']=le.fit_transform(tips['smoker'])

In [13]:
# tips['day']=le.fit_transform(tips['day'])

In [14]:
# tips['time']=le.fit_transform(tips['time'])

In [15]:
#tips.info()

In [16]:
#to_encode=['sex','smoker','day','time']

In [17]:
#for col in to_encode:
    #tips.loc[:,col]=le.fit_transform(tips.loc[:,col])

In [18]:
#tips.info()

In [19]:
#to_dummy =['sex','smoker','day','time']

In [20]:
#to_dummy

In [21]:
#dummies=pd.get_dummies(to_dummy)

In [22]:
#dummies

In [23]:
#tips=pd.concat([dummies,tips],axis=1)
#mirroring effect may result loss of values

tips
day	sex	smoker	time	total_bill	tip	sex	smoker	day	time	size
0	0.0	1.0	0.0	0.0	16.99	1.01	Female	No	Sun	Dinner	2
1	0.0	0.0	1.0	0.0	10.34	1.66	Male	No	Sun	Dinner	3
2	1.0	0.0	0.0	0.0	21.01	3.50	Male	No	Sun	Dinner	3
3	0.0	0.0	0.0	1.0	23.68	3.31	Male	No	Sun	Dinner	2
4	NaN	NaN	NaN	NaN	24.59	3.61	Female	No	Sun	Dinner	4
...	...	...	...	...	...	...	...	...	...	...	...
239	NaN	NaN	NaN	NaN	29.03	5.92	Male	No	Sat	Dinner	3
240	NaN	NaN	NaN	NaN	27.18	2.00	Female	Yes	Sat	Dinner	2
241	NaN	NaN	NaN	NaN	22.67	2.00	Male	Yes	Sat	Dinner	2
242	NaN	NaN	NaN	NaN	17.82	1.75	Male	No	Sat	Dinner	2
243	NaN	NaN	NaN	NaN	18.78	3.00	Female	No	Thur	Dinner	2
244 rows × 11 columns

In [24]:
 from sklearn.preprocessing import OneHotEncoder

In [30]:
ohe=OneHotEncoder(sparse=False)

In [31]:
ohe

OneHotEncoder(categories='auto', drop=None, dtype=<class 'numpy.float64'>,
              handle_unknown='error', sparse=False)

In [32]:
X=tips.loc[:,['sex', 'smoker', 'day', 'time']]

In [33]:
X=ohe.fit_transform(X)

In [34]:
X

array([[1., 0., 1., ..., 0., 1., 0.],
       [0., 1., 1., ..., 0., 1., 0.],
       [0., 1., 1., ..., 0., 1., 0.],
       ...,
       [0., 1., 0., ..., 0., 1., 0.],
       [0., 1., 1., ..., 0., 1., 0.],
       [1., 0., 1., ..., 1., 1., 0.]])

In [35]:
X.shape

(244, 10)

X=pd.Series(X)
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-38-545395147443> in <module>
----> 1 X=pd.Series(X)

~\anaconda3\lib\site-packages\pandas\core\series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    303                     data = data.copy()
    304             else:
--> 305                 data = sanitize_array(data, index, dtype, copy, raise_cast_failure=True)
    306 
    307                 data = SingleBlockManager(data, index, fastpath=True)

~\anaconda3\lib\site-packages\pandas\core\construction.py in sanitize_array(data, index, dtype, copy, raise_cast_failure)
    480     elif subarr.ndim > 1:
    481         if isinstance(data, np.ndarray):
--> 482             raise Exception("Data must be 1-dimensional")
    483         else:
    484             subarr = com.asarray_tuplesafe(data, dtype=dtype)

Exception: Data must be 1-dimensional
